File: ios/CTCodeModule/CTCodeModule.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_REMAP_MODULE (ServiceCodeModule, CTCodeModule, NSObject)
RCT_EXTERN_METHOD(
  getCode: (RCTPromiseResolveBlock)resolve
  rejecter: (RCTPromiseRejectBlock)reject
)
@end

File: ios/CTCodeModule/CTCodeModule.swift
import Foundation

@objc(CTCodeModule)
open class CTCodeModule : NSObject, RCTBridgeModule {
  public static func moduleName() -> String! {
    "ServiceCodeModule"
  }

  @objc
  open func getCode(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
    ) {
      let code: Optional<String> = Bundle.main.object(forInfoDictionaryKey: "service-key") as? String
      resolve(code ?? "")
    }
}
